internal/source/artifact_image.go

```diff
@@ -0,0 +1,49 @@
+package source
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+
+	oras "oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+const unpackLocation = "/tmp/"
+
+type ArtifactImage struct {
+}
+
+func (i *ArtifactImage) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
+	if bundle.Spec.Source.Type != rukpakv1alpha1.SourceTypeOCIArtifacts {
+		return nil, fmt.Errorf("bundle source type %q not supported", bundle.Spec.Source.Type)
+	}
+	if bundle.Spec.Source.Image == nil {
+		return nil, fmt.Errorf("bundle source image configuration is unset")
+	}
+	tmpfs, err := file.New(unpackLocation)
```

Comment:
Do these temp files ever get cleaned up after a certain amount of time? As far as I understand it, Kubernetes treats the `/tmp` directory as an in memory file store on the pod and contributes to it's overall memory consumption. I couldn't find any information on whether or not kube will automatically clean these up after a certain amount of time so I am a bit concerned that we will just be consistently increasing the memory consumption of the pod with this route.

If this is a valid concern, it might be nice to consider using an in memory file store like the [`afero.MemMapFs`](https://pkg.go.dev/github.com/spf13/afero#MemMapFs) which I *think* still implements the Go `fs.FS` interface. I *think* using an implementation like this would allow for the Go garbage collection to automatically free up the memory when it is no longer needed.